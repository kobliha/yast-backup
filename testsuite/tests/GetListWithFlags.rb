# encoding: utf-8

# Backup module - testsuite
#
# GetListWithFlags function tests
#
# testedfiles: backup/functions.ycp
#
# $Id$
#
module Yast
  class GetListWithFlagsClient < Client
    def main
      Yast.import "UI"
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.include self, "backup/functions.rb"

      Testsuite.Dump(GetListWithFlags(nil, nil))
      Testsuite.Dump(GetListWithFlags([], []))
      Testsuite.Dump(GetListWithFlags([], ["/dev"]))
      Testsuite.Dump(GetListWithFlags(["/dev"], []))
      Testsuite.Dump(GetListWithFlags(["/dev", "/etc"], ["/etc"]))

      nil
    end
  end
end

Yast::GetListWithFlagsClient.new.main
