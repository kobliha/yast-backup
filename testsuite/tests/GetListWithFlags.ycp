// Backup module - testsuite
//
// GetListWithFlags function tests
//
// $Id$
//

{
include "testsuite.ycp";

map readmap = $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "is_uml" : false, "cdrom" : $["manual" : []]],
    ];

map execmap = $[ "target" : $["bash_output" : $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]]];

TESTSUITE_INIT([ readmap, $[], execmap ], nil);

include "backup/functions.ycp";

DUMP( GetListWithFlags(nil, nil) );
DUMP( GetListWithFlags([], []) );
DUMP( GetListWithFlags([], ["/dev"]) );
DUMP( GetListWithFlags(["/dev"], []) );
DUMP( GetListWithFlags(["/dev", "/etc"], ["/etc"]));

}

