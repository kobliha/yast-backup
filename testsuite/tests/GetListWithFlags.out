Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Dir	.autoyast2.desktop."s": []
Dir	.yast2.groups.s: []
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp false
Read	.probe.system nil
Read	.sysconfig.console.CONSOLE_ENCODING nil
Execute	.target.bash_output "locale -k charmap" $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]
Read	.target.tmpdir "/tmp"
Read	.probe.floppy.manual nil
Read	.probe.cdrom.manual []
Dump	[]
Dump	[]
Dump	[]
Dump	[`item (`id ("/dev"), "/dev", false)]
Dump	[`item (`id ("/dev"), "/dev", false), `item (`id ("/etc"), "/etc", true)]
